### PR TITLE
snipe-it: convert to buildComposerProject

### PR DESCRIPTION
**Pull request by yayayayaka**
Marked as draft until #261413 has been merged.

## Description of changes
- Removes the updater script
- Builds Snipe-IT with `php.buildComposerProject`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/snipe-it/default.nix [hunk 1]
```diff
@@ -1,52 +1,52 @@
...
+php.buildComposerProject (finalAttrs: {
   pname = "snipe-it";
   version = "6.2.2";
 
   src = fetchFromGitHub {
     owner = "snipe";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "11i9ijkl7am5k48y7r5k6nki2827cd7mw3dr1xj8dvb8diwaskqi";
+    repo = finalAttrs.pname;
```
Comment (drupol) [inline pkgs/servers/web-apps/snipe-it/default.nix:15]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/servers/web-apps/snipe-it/default.nix [hunk 2]
```diff
@@ -1,52 +1,52 @@
...
+    repo = "snipe-it";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-EU+teGxo7YZkD7kNXk9jRyARpzWz5OMRmaWqQ6eMKYY=";
   };
 
-  passthru.tests = nixosTests.snipe-it;
-  passthru.phpPackage = php;
+  vendorHash = "sha256-JcBcrETbjGJFlG1dH/XXqmb9MlKr0ICdnEx7/61Z5io=";
+
+  installPhase = ''
```
Comment (drupol) [inline pkgs/servers/web-apps/snipe-it/default.nix:22]: How about using a `postInstall` hook? So you don't have to copy the files yourself.

**Comment by drupol** [comment]
How it's going with this?

**Comment by yayayayaka** [comment]
aaaaa I got distracted. Anyway, I have implemented your `postInstall` suggestion and updated the module accrodingly.

**Comment by drupol** [comment]
@GrahamcOfBorg build snipe-it

**Comment by yayayayaka** [comment]
Yes. I have deployed it on x86_64-linux for manual testing and the NixOS test has passed.

**Comment by drupol** [comment]
Excellent.

**Review by drupol — APPROVED** [review]
So happy to see this builder being used by more and more contributors :)

**Pull request merged**